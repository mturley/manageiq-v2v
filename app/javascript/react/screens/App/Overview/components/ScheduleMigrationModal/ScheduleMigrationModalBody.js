import React from 'react';
import { Icon } from 'patternfly-react';
import PropTypes from 'prop-types';
import { initializeDatepicker } from '../../helpers';

class ScheduleMigrationModalBody extends React.Component {
  componentDidMount() {
    if (this.props.showDatepicker) {
      const { handleDatepickerChange, defaultDate } = this.props;
      initializeDatepicker(handleDatepickerChange, defaultDate);
    }
  }

  componentDidUpdate(prevProps, prevState, snapshot) {
    if (prevProps.showDatepicker !== this.props.showDatepicker) {
      if (this.props.showDatepicker) {
        const { handleDatepickerChange, defaultDate } = this.props;
        initializeDatepicker(handleDatepickerChange, defaultDate);
      }
    }
  }

  render() {
    const { setScheduleMode, startNowLabel, startLaterLabel, showDatepicker } = this.props;

    const handleRadioChange = event => {
      setScheduleMode(event.currentTarget.value === 'schedule_migration_now');
    };

    return (
      <div>
        <div className="row">
          <div className="col-xs-12">
            <div className="form-group">
              <div className="radio">
                <label style={{ fontWeight: '600' }}>
                  <input
                    type="radio"
                    name="schedule_migration_start"
                    value="schedule_migration_now"
                    onChange={handleRadioChange}
                    checked={!showDatepicker}
                  />
                  {startNowLabel}
                </label>
              </div>
              <div className="radio">
                <label style={{ fontWeight: '600' }}>
                  <input
                    type="radio"
                    name="schedule_migration_start"
                    value="schedule_migration_later"
                    onChange={handleRadioChange}
                    checked={showDatepicker}
                  />
                  {startLaterLabel}
                </label>
              </div>
            </div>
          </div>
        </div>
        {showDatepicker && (
          <div className="row">
            <div className="col-xs-12">
              <div className="form-group">
                <div id="dateTimePicker" className="input-group date-time-picker-pf">
                  <input id="dateTimeInput" type="text" className="form-control" />
                  <span className="input-group-addon">
                    <Icon type="fa" name="calendar" />
                  </span>
                </div>
              </div>
            </div>
          </div>
        )}
      </div>
    );
  }
}

ScheduleMigrationModalBody.propTypes = {
  handleDatepickerChange: PropTypes.func,
  setScheduleMode: PropTypes.func,
  defaultDate: PropTypes.instanceOf(Date),
  startNowLabel: PropTypes.string,
  startLaterLabel: PropTypes.string,
  showDatepicker: PropTypes.bool
};

export default ScheduleMigrationModalBody;
