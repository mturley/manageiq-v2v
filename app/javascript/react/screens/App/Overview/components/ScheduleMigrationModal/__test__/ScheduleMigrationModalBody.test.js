import React from 'react';
import { shallow } from 'enzyme';
import ScheduleMigrationModalBody from '../ScheduleMigrationModalBody';

describe('When correctly initialized', () => {
  const baseProps = {
    handleDatepickerChange: jest.fn(),
    setScheduleMode: jest.fn(),
    defaultDate: new Date('2020-01-01 10:00:00'),
    startNowLabel: 'start now label',
    startLaterLabel: 'start later label',
    showDatepicker: false
  };

  test('renders schedule modal body', () => {
    const wrapper = shallow(<ScheduleMigrationModalBody {...baseProps} />);
    expect(wrapper).toMatchSnapshot();
  });
});
