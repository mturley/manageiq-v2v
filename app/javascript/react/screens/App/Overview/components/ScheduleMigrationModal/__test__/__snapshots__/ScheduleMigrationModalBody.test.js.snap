// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`When correctly initialized renders schedule modal body 1`] = `
<div>
  <div
    className="row"
  >
    <div
      className="col-xs-12"
    >
      <div
        className="form-group"
      >
        <div
          className="radio"
        >
          <label
            style={
              Object {
                "fontWeight": "600",
              }
            }
          >
            <input
              checked={true}
              name="schedule_migration_start"
              onChange={[Function]}
              type="radio"
              value="schedule_migration_now"
            />
            start now label
          </label>
        </div>
        <div
          className="radio"
        >
          <label
            style={
              Object {
                "fontWeight": "600",
              }
            }
          >
            <input
              checked={false}
              name="schedule_migration_start"
              onChange={[Function]}
              type="radio"
              value="schedule_migration_later"
            />
            start later label
          </label>
        </div>
      </div>
    </div>
  </div>
</div>
`;
